internal/common/update/patch_payload.go

```diff
@@ -129,15 +155,19 @@ func convertJSONDiffToJSONPatch(patch jsondiff.Patch) (jsonpatch.Patch, error) {
 // 4. Adds nested "removed" values from the state to the request
 // 5. Use `jsonpatch` to apply each attribute plan & state patch to an empty JSON object
 // 6. Create a `patchReq` pointer with the final JSON object marshaled to `T` or return nil if there are no changes (`{}`)
-func PatchPayload[T any](state, plan *T) (*T, error) {
+func PatchPayload[T any](state, plan *T, options ...PatchOptions) (*T, error) {
+	return PatchPayloadDiffTypes(state, plan, options...)
+}
+
+func PatchPayloadDiffTypes[T any, U any](state *T, plan *U, options ...PatchOptions) (*U, error) {
```

Comment:
are both PatchPayloadDiffTypes and PatchPayload needed?

Reply:
Good point!
Removed DiffTypes and changed the PatchPayload signature